### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by lfmnovaes**
## Changes
- Applied DRY principles
- Checked if other principles are respected (AGNI, KISS, and Ruby style)
- Followed SOLID principles.
- Used `main.rb` is only to start the code and ask the user for the initial choice of options (or to quit).
- Extracted some pieces of logic into a separate class (`choice.rb`)

### CODE REVIEW

File: choice.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      return
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+    pp = gets.chomp
+    @people.push(Student.new(name: name, age: age, parent_permission: translate_answer(pp.downcase)))
+  end
```
Comment (omar-labana) [inline choice.rb]: - [x] CRUD operations and UI rendering for different classes are two separate features that shouldn't coexist in the same class. I'd suggest revisiting this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) and making sure that you are following the `Single responsibility rule`. Kindly do the same for other CRUD operations as well.
> A class should have one and only one reason to change, meaning that a class should have only one job.

File: choice.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      puts 'Not a valid option'
+      return
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+    pp = gets.chomp
+    @people.push(Student.new(name: name, age: age, parent_permission: translate_answer(pp.downcase)))
```
Comment (omar-labana) [inline choice.rb]: - [x] This line (Student.new(name: name, age: age, parent_permission: translate_answer(pp.downcase))) should've been placed in the 'create_person()'`
> A client should never be forced to implement an interface that it doesn’t use, or clients shouldn’t be forced to depend on methods they do not use.

File: choice.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people.push(Teacher.new(name: name, age: age, specialization: specialization))
```
Comment (omar-labana) [inline choice.rb]: - [x] This line (Teacher.new(name: name, age: age, parent_permission: translate_answer(pp.downcase))) should've been placed in the 'create_person()'`
> A client should never be forced to implement an interface that it doesn’t use, or clients shouldn’t be forced to depend on methods they do not use.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @lfmnovaes and @mike2611, 

Excellent work so far! 👌
You refactored your code neatly and nicely. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @lfmnovaes,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**